src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageLoader.cs

```diff
@@ -55,7 +55,8 @@ public async Task<bool> DoesPackageSupportTargetFrameworkAsync(NuGetReference pa
             }
 
             var packageFrameworks = await GetTargetFrameworksAsync(packageArchive, token).ConfigureAwait(false);
-            return packageFrameworks.Any(f => DefaultCompatibilityProvider.Instance.IsCompatible(NuGetFramework.Parse(targetFramework.Name), f));
+
+            return packageFrameworks.Any(f => targetFrameworks.All(targetFramework => DefaultCompatibilityProvider.Instance.IsCompatible(NuGetFramework.Parse(targetFramework.Name), f)));
```

Comment:
This isn't right. This requires that the package support a single TFM that works with all of the project's TFMs. Instead, we need to make sure that for all the project's TFMs, there's at least one package TFM that will work for it.

For example, if a project targets net472;net5 and a NuGet package has target frameworks net461 and netcoreapp3.1, then this method should return true (since that project can use that package) but I think currently it will return false since none of the package frameworks work for all of the project frameworks.

Reply:
Good catch.

Reply:
This is a hard one for a unit test due to the nuget package dependency :(